### ISSUE TITLE
Is this a bug?

### ISSUE THREAD

**Issue by albus01** [SlyMarbo/spdy#90]
When I loop sending query to get a small html long time,I can get one err like this:
(spdy) 2015/07/14 17:52:08 error_handling.go:17: Error: Received PROTOCOL_ERROR on stream 0. Closing connection.
RoundTripper returned a response & error; ignoring response

And if I get two errors above continuous,the client will block cause the DefaultMaxIdleConnsPerHost is 2,and when occur this error,not produce t.connLimit[req.URL.Host].And it will block at dial() function at line 108:<-t.connLimit[u.Host]

And why the server will response this error?Does some error occured in the client?

**Comment by SlyMarbo** [comment]
This should hopefully be fixed in 65e8ed3. Is this library being used in the server too? It's difficult to say what's going on without further context.

**Comment by albus01** [comment]
Yes,the lib is used in the server.But has some bugs and have fixed.My friends has commit some patches to you.This 65e8ed3 has fixed the block issue,but not fix bugs in source.If I use this client to test the server's qps,the server close the connection more often and client will rebuild the connection.
